src/move/econia/sources/user.move

```diff
@@ -409,6 +409,59 @@ module econia::user {
         size: u64
     }
 
+    /// Table keys are market account IDs.
+    struct MarketEventHandles has key {
+        fill_events: Tablist<u128, EventHandle<FillEvent>>,
+        place_limit_order_events:
+            Tablist<u128, EventHandle<PlaceLimitOrderEvent>>,
+        place_market_order_events:
+            Tablist<u128, EventHandle<PlaceMarketOrderEvent>>
+    }
+
+    struct FillEvent has copy, drop, store {
+        market_id: u64,
+        size: u64,
+        price: u64,
+        maker_side: bool,
+        maker: address,
+        maker_custodian_id: u64,
+        maker_order_id: u128,
+        taker: address,
+        taker_custodian_id: u64,
+        taker_order_id: u128
+    }
+
+    struct PlaceLimitOrderEvent has copy, drop, store {
+        market_id: u64,
+        user: address,
+        custodian_id: u64,
+        integrator: address,
+        side: bool,
+        size: u64,
+        price: u64,
+        restriction: u8,
+        self_match_behavior: u8,
+        base_traded_as_taker: u64,
+        quote_traded_as_taker: u64,
+        quote_fee_paid_as_taker: u64,
+        size_posted_as_maker: u64,
+        order_id: u128
+    }
```

Comment:
The initial fill upon placement should not be considered part of the placement—the order is placed, and then immediately filled if it crosses the book. `base_traded_as_taker`, `quote_traded_as_taker`, and `quote_fee_paid_as_taker` can all be part of a fill event that is emitted after the place limit order event. The nuances are important here.

Reply:
Fees broken out into proposed `TakerFillSummaryEvent` per https://github.com/econia-labs/econia/pull/321#issuecomment-1626162868